src/styles/blocks/_calendar.scss

```diff
@@ -0,0 +1,66 @@
+.calendar {
+  width: $width-calendar + 6px;
+
+  @for $i from 28 to $day-count {
+    &--month-length-#{$day-count} {
+      .calendar__day {
+        &:nth-last-child(-n+#{$remaining-days}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  $start-day: 
+    "mon" $width-calendar - (7 * $width-day),
+    "tue" $width-calendar - (6 * $width-day) + 1px,
+    "wed" $width-calendar - (5 * $width-day) + 2px,
+    "thu" $width-calendar - (4 * $width-day) + 3px,
+    "fri" $width-calendar - (3 * $width-day) + 4px,
+    "sat" $width-calendar - (2 * $width-day) + 5px,
+    "sun" $width-calendar - (1 * $width-day) + 6px;

```

Comment:
you can put gap value in variable and calculate necessary margin and better to keep all variables in separate file and something strange in formula I suppose you should reverse multiplier